### PR TITLE
Increase verbosity of the dogfood runtest.sh

### PR DESCRIPTION
**Pull request by jvillal-amp**
Use 'set -x' to show the content of the lines being executed.

Add setting of PS4 so that 'set -x' will show the filename, function
name, and line number of the line being executed.

### CODE REVIEW

**Comment by StykMartin** [comment]
I guess, we can drop there `-x`, not sure if we really wanna introduce `PS4`.
@danrodrig thoughts?

File: dogfood/runtest.sh [hunk 1]
```diff
@@ -17,6 +17,10 @@
 
 . /usr/bin/rhts-environment.sh
 
+# Make things more verbose
+set -x
+export PS4='+ ${BASH_SOURCE:-}:${FUNCNAME[0]:-}:L${LINENO:-}:   '
```
Comment (StykMartin) [inline dogfood/runtest.sh:22]: We should be fine without `PS4`

**Review by StykMartin — CHANGES_REQUESTED** [review]

**Comment by danrodrig** [comment]
PS4 looks interesting, although not sure how much are we gaining. I usually don't need this info, but it may depend on the script.
I'm OK with adding -x and leaving PS4. We can revisit it later if needed.

**Comment by jvillal-amp** [comment]
> 
> 
> I guess, we can drop there `-x`, not sure if we really wanna introduce `PS4`.

I'll be honest that I am a fan of PS4. Having the line # and the file name can often be crucial to quickly figuring out where something is broken.

An example output line from an OpenStack usage:
`+ ./safe-devstack-vm-gate-wrap.sh:main:L547:   RETVAL=1`

So this tells you that we are in the 'safe-devstack-vm-gate-wrap.sh' file, function 'main', and on line 547. The command being executed is 'RETVAL=1'.

Without PS4 being set all that would be shown is 'RETVAL=1'

Info on this PS4 variable found at:
http://wiki.bash-hackers.org/scripting/debuggingtips

**Comment by StykMartin** [comment]
> > I guess, we can drop there `-x`, not sure if we really wanna introduce `PS4`.
> 
> I'll be honest that I am a fan of PS4. Having the line # and the file name can often be crucial to quickly figuring out where something is broken.
> 
> An example output line from an OpenStack usage:
> `+ ./safe-devstack-vm-gate-wrap.sh:main:L547: RETVAL=1`
> 
> So this tells you that we are in the 'safe-devstack-vm-gate-wrap.sh' file, function 'main', and on line 547. The command being executed is 'RETVAL=1'.
> 
> Without PS4 being set all that would be shown is 'RETVAL=1'
> 
> Info on this PS4 variable found at:
> http://wiki.bash-hackers.org/scripting/debuggingtips

I mean I do understand how the output will look like.. but, do we truly need this for dogfood?

**Comment by jvillal-amp** [comment]
> 
> 
> > > I guess, we can drop there `-x`, not sure if we really wanna introduce `PS4`.
> > 
> > 
> > I'll be honest that I am a fan of PS4. Having the line # and the file name can often be crucial to quickly figuring out where something is broken.
> > An example output line from an OpenStack usage:
> > `+ ./safe-devstack-vm-gate-wrap.sh:main:L547: RETVAL=1`
> > So this tells you that we are in the 'safe-devstack-vm-gate-wrap.sh' file, function 'main', and on line 547. The command being executed is 'RETVAL=1'.
> > Without PS4 being set all that would be shown is 'RETVAL=1'
> > Info on this PS4 variable found at:
> > http://wiki.bash-hackers.org/scripting/debuggingtips
> 
> I mean I do understand how the output will look like.. but, do we truly need this for dogfood?

Of course not :)  Haven't had it up to this point so obviously not needed. But does it hurt anything having more data to assist in the future?

**Comment by StykMartin** [comment]
I mean, we are emitting huge about of data already in dogfood.
Sometimes we are even emitting things that we shouldn't in my opinion.

**Comment by danrodrig** [comment]
> Without PS4 being set all that would be shown is 'RETVAL=1'

I mentioned that I usually don't need this information because you can get the idea by looking at the previous executed lines. But I understand that depends on the script.

> Sometimes we are even emitting things that we shouldn't in my opinion.

Is -x really needed then? The option is fine for debugging the script while writing it, but if you need to rely on that information after the script is done is probably because your script is lacking output.

If not, I would rather keep the output cleaner and improve it as needed in the future.

**Comment by StykMartin** [comment]
I will close this for now @jvillal-amp.
We can reevaluate this later on, if we consider this as needed.

**Comment by jvillal-amp** [comment]
Martin had requested I put this here:

So the reason I proposed this is because it might have helped me to debug things faster when I was working on trying to figure out what was wrong. I eventually was able to figure out what was happening. But just took a bit more time.

I had assumed at first that IntegrationTests/runtest.sh was being run. But I eventually figured out it was beaker-meta-tasks-dogfood. Then trying to figure out which one of the nose tests is being run.

If this had been in there I would have figured it out much quicker and likely saved myself 1-2 hours of time. I know now, but for someone unfamiliar with things I had to spend more time figuring things out.

**Pull request closed without merging**